### PR TITLE
fix: Gcc dump types

### PR DESCRIPTION
**Pull request by dkm**
While doing some cleaning, the typing of the GCC dump feature was found incorrect. This change tries to fix most of it... But some details still need to be fixed (FIXME notes left in the code).

Signed-off-by: Marc Poulhiès <dkm@kataplop.net>

### CODE REVIEW

**Review by RubenRBS — APPROVED** [review]
We can talk about the tests once we look deeper into those, but this looks great, thanks a lot!

**Comment by dkm** [comment]
Thanks for the quick review!

**Comment by dkm** [comment]
this is live :rocket:

File: static/panes/gccdump-view.ts [hunk 1]
```diff
@@ -77,11 +76,7 @@ export class GccDump extends MonacoPane<monaco.editor.IStandaloneCodeEditor, Gcc
     cursorSelectionThrottledFunction: ((e: any) => void) & _.Cancelable;
     selectedPass: string | null = null;
 
-    constructor(hub: Hub, container: Container, state: GccDumpState & MonacoPaneState) {
```
Comment (dkm) [inline static/panes/gccdump-view.ts:80]: possible error in init order?

**Pull request merged**